packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -328,7 +329,12 @@ export function TransferPanelMain({
   const { l1, l2, isConnectedToArbitrum, isSmartContractWallet } =
     useNetworksAndSigners()
 
-  const { provider } = useWallet()
+  const { switchNetwork } = useSwitchNetwork({
+    throwForSwitchChainNotSupported: true,
```

Comment:
Maybe we can refactor this snippet to be shared across all 5 instances it's being duplicated? 
We can have an additional flag for the extra param for `isSwitchingNetworkBeforeTx: true`. Wdyt?

Reply:
i wouldn't want to wrap wagmi hooks to make it hard to trace and change the config
as it's pretty straightforward now for people to see the hook and go to wagmi to find its documentation